src/MagicalClass.java

```diff
@@ -0,0 +1,100 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class MagicalClass extends PlayableCharacter {
+
+ 	// Attributes
+	private static final int maxMana = 100;
+	private int mana;
+	private int inteligence;
+	protected CRUD<Spell> spellList; // CRUD<Spell>
+	protected static final String errorMessage = "Error: Invalid value";
+
+	//Constructors
+	public MagicalClass(String name, int mana, int inteligence) throws Exception {
+		super(name);
+		this.setMana(mana);
+		this.setInteligence(inteligence);
+		this.spellList.initialize();
+	}
+
+ 	public MagicalClass(
+ 			byte level,
+ 			int experience,
+ 			int mana,
+ 			int inteligence,
+ 			String name
+ 	) throws Exception {
+		super(level,name,experience);
+		this.setMana(mana);
+		this.setInteligence(inteligence);
+		this.spellList.initialize();
+	}
+ 	
+ 	// Methods
+ 	public abstract void lernNewSpell(Spell newSpell);
+ 	public abstract void updateSkill(Spell spell);
+
+	public void forgetSkill(int id) {
+		try {
+			this.spellList.delete(id);
+        }catch (Exception error){
+            throw error;
+        }
+	}
+	
+	// Override equals method.
+	@Override
+	public boolean equals(Object obj) {
+        return super.equals(obj);
+    }
+	
+	// Override hashCode method.
+	@Override
+	public int hashCode() {
+		return super.hashCode();
+	}
+
+	// Override toString method.
+	@Override
+	public String toString() {
+		String base = super.toString();
+		String messageFmt = "\n=> MagicalClass\n[Mana: %d, Inteligence: %d]";
+		return base.concat(String.format(messageFmt, getMana(), getInteligence()));
```

Comment:
Create a method showAllSpells to call it from to string and show the toString of each spell